pkg/operator/controller/controller_cabundle_configmap.go

```diff
@@ -0,0 +1,204 @@
+	var configmapNames []string
+	transportConfig := dns.Spec.UpstreamResolvers.TransportConfig
+	if transportConfig.Transport == operatorv1.TLSTransport && transportConfig.TLS != nil && transportConfig.TLS.CABundle.Name != "" {
+		configmapNames = append(configmapNames, transportConfig.TLS.CABundle.Name)
+	}
+	for _, server := range dns.Spec.Servers {
+		transportConfig := server.ForwardPlugin.TransportConfig
+		if transportConfig.Transport == operatorv1.TLSTransport && transportConfig.TLS != nil && transportConfig.TLS.CABundle.Name != "" {
+			configmapNames = append(configmapNames, transportConfig.TLS.CABundle.Name)
+		}
+	}
+
+	var errs []error
+	for _, name := range configmapNames {
+		sourceName := types.NamespacedName{
+			Namespace: GlobalUserSpecifiedConfigNamespace,
+			Name:      name,
+		}
+		haveSource, source, err := r.currentCABundleConfigMap(sourceName)
+		if err != nil {
+			errs = append(errs, fmt.Errorf("failed to get source ca bundle config map %s: %v", sourceName.Name, err))
+			continue
+		}
+
+		destName := CABundleConfigMapName(source.Name)
+		have, current, err := r.currentCABundleConfigMap(destName)
+		if err != nil {
+			errs = append(errs, fmt.Errorf("failed to get destination ca bundle config map %s: %v", destName.Name, err))
+			continue
+		}
+
+		want, desired, err := desiredCABundleConfigMap(dns, haveSource, source, destName)
+		if err != nil {
+			errs = append(errs, fmt.Errorf("failed to generate desired ca bundle config map: %v", err))
+			continue
+		}
+
+		switch {
+		case !want && !have:
+			continue
+		case !want && have:
+			if err := r.client.Delete(context.TODO(), current); err != nil {
+				if !errors.IsNotFound(err) {
+					errs = append(errs, fmt.Errorf("failed to delete configmap: %v", err))
+					continue
+				}
+			} else {
+				logrus.Infof("deleted configmap %s/%s", current.Namespace, current.Name)
+			}
+		case want && !have:
+			if err := r.client.Create(context.TODO(), desired); err != nil {
+				errs = append(errs, fmt.Errorf("failed to create configmap: %v", err))
+				continue
+			}
+			logrus.Infof("created configmap %s/%s", desired.Namespace, desired.Name)
+			if _, _, err = r.currentCABundleConfigMap(destName); err != nil {
+				errs = append(errs, fmt.Errorf("failed to get destination ca bundle configmap %s: %v", destName.Name, err))
+			}
+		case want && have:
+			if updated, err := r.updateCABundleConfigMap(current, desired); err != nil {
+				errs = append(errs, fmt.Errorf("failed to update configmap: %v", err))
+			} else if updated {
+				logrus.Infof("updated configmap %s/%s", desired.Namespace, desired.Name)
+				if _, _, err = r.currentCABundleConfigMap(destName); err != nil {
+					errs = append(errs, fmt.Errorf("failed to get destination ca bundle configmap %s: %v", destName.Name, err))
+				}
+			}
+		}
+	}
+
+	// remove ca bundle configmaps that are not referred in dns anymore.
+	cmSelector, err := metav1.LabelSelectorAsSelector(&cmLabelSelector)
+	if err != nil {
+		return err
```

Comment:
I agree with returning here, but, another edge case. What if you had errors above and this error'ed too? You'd lose your errors from above.
Consider:
```suggestion
		errs = append(errs, err)
                return utilerrors.NewAggregate(errs) 
```

Reply:
Ah I missed this one. Thanks!